Composer/packages/electron-server/src/splash/splashScreen.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { BrowserWindow, systemPreferences } from 'electron';
+
+import { getSplashScreenContent, statusElmId } from './template';
+
+export type SplashScreenProps = {
+  getMainWindow: () => BrowserWindow | undefined;
+  color?: string;
+  icon?: string;
+  width?: number;
+  height?: number;
+  productName?: string;
+  productFamily?: string;
+  logo?: string;
+  website?: string;
+  status?: string;
+};
+
+export const initSplashScreen = async ({
+  getMainWindow,
+  color: initColor,
+  icon,
+  width = 600,
+  height = 400,
+  productName,
+  productFamily,
+  logo,
+  website,
+  status,
+}: SplashScreenProps) => {
+  // If no color is provided, uses OS accent color
+  const color = initColor || (systemPreferences.getAccentColor && `#${systemPreferences.getAccentColor()}`);
+
+  const splashScreenWindow = new BrowserWindow({
+    parent: getMainWindow(),
+    show: false,
+    width,
+    height,
+    modal: true,
+    transparent: true,
+    skipTaskbar: true,
+    frame: false,
+    autoHideMenuBar: true,
+    alwaysOnTop: true,
+    resizable: false,
+    movable: false,
+    icon,
+    webPreferences: {
+      webSecurity: false,
+    },
+  });
+
+  const args = {
+    productName,
+    productFamily,
+    logo,
+    website,
+    color,
+    status,
+  };
+
+  splashScreenWindow.on('ready-to-show', () => {
+    splashScreenWindow.show();
+  });
+
+  const file = 'data:text/html;charset=UTF-8,' + encodeURIComponent(getSplashScreenContent(args));
+  await splashScreenWindow.loadURL(file);
+
+  const startApp = () => {
+    setTimeout(() => splashScreenWindow.destroy(), 500);
```

Comment:
Any chance that splashScreenWindow could be undefined/null here after the timeout?  Should it be set to undefined after being destroyed?

Reply:
1. It's guaranteed to be "not-null"
2. That's a good code practice if it was exposed to the caller side, but since no one else is going to consume the window, it's fine.